src/main/java/picard/sam/ReorderSam.java

```diff
@@ -117,7 +117,7 @@ protected int doWork() {
 
         log.info("Writing reads...");
         if (in.hasIndex()) {
-            final SAMFileWriter out = new SAMFileWriterFactory().makeSAMOrBAMWriter(outHeader, true, OUTPUT);
+            final SAMFileWriter out = new SAMFileWriterFactory().makeWriter(outHeader, true, OUTPUT, REFERENCE_SEQUENCE);
```

Comment:
For some reason (probably because it was a convenient way to override the optional attribute for REFERENCE for this tool) ReorderSam has a duplicate command line arg for reference. There is one in the ReorderSam class called REFERENCE, and one inherited from CommandLineProgram called REFERENCE_SEQUENCE. Both of them are used in the tool code, but only REFERENCE is properly populated by the command line parser. This code should be fixed so the variable usage is consistent. It might be best to just remove the local REFERENCE variable and rely on the inherited REFERENCE_SEQUENCE one, and just have the code throw if the argument is not populated. I think that would retain the behavior and remove the ambiguity.